File: Java/Medium/Accepted/display-table-of-food-orders-in-a-restaurant_1.java
class Solution {
    public List<List<String>> displayTable(List<List<String>> orders) {
        
        Set<String> items = new HashSet<>();
        Map<Integer, Map<String, Integer>> map = new TreeMap<>();
        for (List<String> order : orders) {
            int table = Integer.parseInt(order.get(1));
            String food = order.get(2);
            items.add(food);
            Map<String, Integer> tableMap = map.getOrDefault(table, new HashMap<>());
            tableMap.put(food, tableMap.getOrDefault(food, 0) + 1);
            map.put(table, tableMap);
        }
        List<List<String>> res = new ArrayList<>();
        List<String> header = new ArrayList<>();
        header.add("Table");
        List<String> itemsList = new ArrayList<>(items);
        Collections.sort(itemsList);
        header.addAll(itemsList);
        res.add(header);
        for (int table : map.keySet()) {
            List<String> row = new ArrayList<>();
            row.add(String.valueOf(table));
            for (String item : itemsList) {
                row.add(String.valueOf(map.get(table).getOrDefault(item, 0)));
            }
            res.add(row);
        }
        return res;
    }
}

File: C++/Medium/Accepted/display-table-of-food-orders-in-a-restaurant_1.cpp
class Solution {
public:
    vector<vector<string>> displayTable(vector<vector<string>>& orders) {
        
        map<int, map<string, int>> m;
        set<string> food;
        for (auto order : orders) {
            m[stoi(order[1])][order[2]]++;
            food.insert(order[2]);
        }
        vector<vector<string>> res;
        vector<string> header = {"Table"};
        for (auto f : food) {
            header.push_back(f);
        }
        res.push_back(header);
        for (auto it = m.begin(); it != m.end(); it++) {
            vector<string> row = {to_string(it->first)};
            for (auto f : food) {
                row.push_back(to_string(it->second[f]));
            }
            res.push_back(row);
        }
        return res;
    }
};